src/go/configgenerator/filtergen/http_connection_manager.go

```diff
@@ -0,0 +1,267 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filtergen
+
+import (
+	"fmt"
+
+	"github.com/GoogleCloudPlatform/esp-v2/src/go/options"
+	"github.com/GoogleCloudPlatform/esp-v2/src/go/tracing"
+	"github.com/GoogleCloudPlatform/esp-v2/src/go/util"
+	"github.com/GoogleCloudPlatform/esp-v2/src/go/util/httppattern"
+	acpb "github.com/envoyproxy/go-control-plane/envoy/config/accesslog/v3"
+	corepb "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	routepb "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
+	facpb "github.com/envoyproxy/go-control-plane/envoy/extensions/access_loggers/file/v3"
+	hcmpb "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/http_connection_manager/v3"
+	"github.com/golang/glog"
+	servicepb "google.golang.org/genproto/googleapis/api/serviceconfig"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/anypb"
+	"google.golang.org/protobuf/types/known/structpb"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+)
+
+const (
+	// HTTPConnectionManagerFilterName is the Envoy filter name for debug logging.
+	HTTPConnectionManagerFilterName = "envoy.filters.network.http_connection_manager"
+)
+
+type HTTPConnectionManagerGenerator struct {
```

Comment:
Why not put HCM under `configgenerator` folder not `configgenerator/filter` folder? Will we have different HCM HCM configgenrator inside g3? 

HCM containing all http filters is a one-level lower than those in `filter` folder so it is wired to put them together.

Reply:
We will have a different HCM in google3. Remember you added custom local reply config for our use case? I removed that from ESPv2, but we will override it internally.

I would like to reuse some of the test utilities, like `filtergentest`. Not sure if moving it out of `filtergen` folder is good idea then.

In the future if we have more network filters, I can separate `filtergen` into `httpfiltergen` and `networkfiltergen`